ioeffect/src/main/scala/scalaz/ioeffect/IO.scala

```diff
@@ -704,9 +699,12 @@ object IO extends IOInstances {
    * }}}
    */
   final def syncThrowable[A](effect: => A): IO[Throwable, A] =
-    syncCatch(effect) {
-      case t: Throwable => t
-    }
+    IO.absolve(
```

Comment:
it's not applying a partial function

Reply:
Instead of this duplication you can improve `syncCatch` with this

```scala
catch f andThen (Left[E, A](_))
```

That way the PF is evaluated only once

Reply:
I came up with something using `suspend`, wdyt?